### PR TITLE
Blockchain cleanup

### PR DESCRIPTION
**Pull request by hackaugusto**
Review after #1112 

Split the raiden.network.rpc.client module

### CODE REVIEW

**Comment by LefterisJP** [comment]
rebased on master

File: raiden/network/rpc/filters.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        json_data['topics'] = [
+            topic_encoder(topic)
+            for topic in topics
+        ]
+
+    return jsonrpc_client.call('eth_newFilter', json_data)
+
+
+def get_filter_events(jsonrpc_client, contract_address, topics, from_block=None, to_block=None):
+    """ Custom new filter implementation to handle bad encoding from geth rpc. """
```
Comment (LefterisJP) [inline raiden/network/rpc/filters.py]: `new filter` -> `get filter`

File: raiden/network/proxies/README.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  implementations. [tests/utils/*_client.py]
+- Expose a synchronous interface by default
+  - poll for the transaction hash
+  - check if the proper events were emited
+  - use `call` and `transact` to interact with client proxies
+- Check errors:
+  - `call` returns the empty string if the target smart contract does not
+  exist or the call throws, handle it accordingly (there is no way to
+  distinguish a function that returns the empty string from the error)
+  - the smart contract executed with a `transact` may fail with a throw, this
```
[the remark this answers is not in the document]
Comment (hackaugusto) [reply raiden/network/proxies/README.txt]: asserts are using throws, requires are using reverts:

https://solidity.readthedocs.io/en/develop/control-structures.html?highlight=throw#error-handling-assert-require-revert-and-exceptions

[the remark this answers is not in the document]
Comment (LefterisJP) [reply raiden/network/proxies/README.txt]: yep. But still asserts are going to be the most rare occurence as they are designed to never happen unless there is an internal error/broken invariant. In all other cases (even the marked as deprecated `throw;` statement) will end up being a revert.

Perhaps tweak the text to make that evident?

[the remark this answers is not in the document]
Comment (hackaugusto) [reply raiden/network/proxies/README.txt]: done

**Pull request merged**